src/psyclone/domain/lfric/lfric_arg_descriptor.py

```diff
@@ -0,0 +1,676 @@
+        if len(arg_type.args) == 4:
+            try:
+                from psyclone.parse.kernel import get_stencil, get_mesh
+                if "stencil" in str(arg_type.args[3]):
+                    self._stencil = get_stencil(
+                        arg_type.args[3],
+                        LFRicArgDescriptor.VALID_STENCIL_TYPES)
+                elif "mesh" in str(arg_type.args[3]):
+                    self._mesh = get_mesh(arg_type.args[3],
+                                          LFRicArgDescriptor.VALID_MESH_TYPES)
+                else:
+                    raise ParseError("Unrecognised metadata entry")
+            except ParseError as err:
+                raise ParseError(
+                    "In the LFRic API the 4th argument of a 'meta_arg' "
+                    "field entry must be either a valid stencil specification"
+                    "or a mesh identifier (for inter-grid kernels). However, "
+                    "entry '{0}' raised the following error: {1}".
+                    format(arg_type, str(err)))
+
+        # Test allowed accesses for fields
+        field_disc_accesses = [AccessType.READ, AccessType.WRITE,
+                               AccessType.READWRITE]
+        # Convert generic access types to GH_* names for error messages
+        api_config = Config.get().api_conf(API)
+        rev_access_mapping = api_config.get_reverse_access_mapping()
+
+        # Check fields on discontinuous function spaces
+        fld_disc_acc_msg = [rev_access_mapping[acc] for acc in
+                            field_disc_accesses]
+        if self._function_space1.lower() in \
+           FunctionSpace.VALID_DISCONTINUOUS_NAMES \
+           and self._access_type not in field_disc_accesses:
+            raise ParseError(
+                "In the LFRic API allowed accesses for a field on a "
+                "discontinuous function space ('{0}') are {1}, but found "
+                "'{2}' in '{3}'".
+                format(self._function_space1.lower(), fld_disc_acc_msg,
+                       rev_access_mapping[self._access_type], arg_type))
+
+        # Check fields on continuous function spaces
+        # TODO in issues #471 and #138: Allowed accesses for fields on
+        # continuous function spaces for the user-defined kernels that loop
+        # over cells should only be [AccessType.READ, AccessType.INC]. This
+        # will remove the currently implicitly allowed "WRITE" access.
+        # The kernels that loop over DoFs (built-ins) and are specified on
+        # 'ANY_SPACE_*' will have the same allowed accesses as discontinuous
+        # fields as they do not need colouring when updating quantities
+        # in parallel.
+        if self._function_space1.lower() in \
+           FunctionSpace.CONTINUOUS_FUNCTION_SPACES \
+           and self._access_type == AccessType.READWRITE:
+            raise ParseError(
+                "It does not make sense for a field on a continuous "
```

Comment:
Yes, you are correct. I have rephrased the message and included the list of allowed accesses. Please note that the list for now includes `WRITE` which should not technically be used for `ANY_SPACE` except for the built-in kernels (looping over DoFs). However, there are still kernels on continuous and any spaces that have `WRITE` - this will be dealt with in #138 and #471 after this pull request.